app/controllers/desks_controller.rb

```diff
@@ -23,6 +23,8 @@ def show
 
   def update
     if @desk.update(desk_params)
+      changes = @desk.changes_to_broadcast
```

Comment:
I'd use `desk_params` since we only permit `name` and `occupied` to be updated and get rid of the `changes_to_broadcast` method

Reply:
What if `desk_params` sets `occupied` to false despite it already being false. Then we will broadcast a change that's not a change. The point of `changes_to_broadcast` is to determine if something actually changed.

Reply:
In that case, I'd move the `changes_to_broadcast` method to the controller since broadcasting websocket events is the job of the controller and not the model :)

Reply:
I agree that the model shouldn't broadcast on websockets but this method is only responsible for returning model data in a specific format. The broadcasting is still happening in the controller.

As far as the placement of such a method goes, I think this falls under the ["Fat Model, Skinny Controller"](https://www.sitepoint.com/10-ruby-on-rails-best-practices/) best practice for rails.

Reply:
> this method is only responsible for returning model data in a specific format

To me the end result of the method isn't as important as the reason why the method exists in the first place. The method was created so that the controller can determine whether or not it sends the websocket event.

> I think this falls under the "Fat Model, Skinny Controller

I agree with the pattern but that doesn't mean that any method related to model data should go inside the model.

The job of updating the model is the job of the controller. And furthermore, the job of broadcasting the websocket event, which is the main purpose of having the method after all, is also the job of the controller.

From the site you linked, `non-response-related logic should go in the model`, to me this method is very much related the response since it updates the UI if this method returns something.